app/lib/account_deletion_config.rb

```diff
@@ -1,27 +1,38 @@
 # frozen_string_literal: true
 
-module AccountDeletionConfig
-  module_function
+class AccountDeletionConfig
+  def self.configure(config)
+    @configuration = new(config)
+  end
+
+  def self.config
+    (@configuration || new).config
+  end
 
-  def config
-    @config ||= load_config
+  def self.valid?
+    (@configuration || new).valid?
   end
 
-  def valid?
-    instance_variable_defined?(:@valid) ? @valid : (@valid = check_validity)
+  def initialize(config = {})
+    format_config(config)
+    check_validity(config)
   end
 
-  def load_config
-    config = ThreeScale.config.features.account_deletion.each_with_object({}) do |(key, value), collection|
+  attr_reader :config, :valid
+  alias valid? valid
+
+  private
+
+  def format_config(config)
+    config = (config || {}).each_with_object({}) do |(key, value), collection|
       collection[key.to_sym] = value if value.is_a?(Integer)
     end
-    ActiveSupport::OrderedOptions.new.merge(config)
+    @config = ActiveSupport::OrderedOptions.new.merge(config)
   end
 
-  def check_validity
-    valid = %i[account_suspension account_inactivity contract_unpaid_time].all? { |key| config.key?(key) }
-    return valid if valid || ThreeScale.config.features.account_deletion.blank?
+  def check_validity(initial_config)
```

Comment:
You can use the it without parameter. `@configuration` is defined

Reply:
@hallelujah Those 2 are not the same. `@config` does not contain the incorrect params because it has been filtered, but `initial_config` has them, and I need to know if the params are incorrect or not set at all to show the warning or not 😄